### PR TITLE
Model Classes

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Creating models and associations
The following changes are made in this branch:
- Created model classes: `User`, `Like`, `Comment` & `Post`
- Added associations between models based on database schema
- Implemented the following methods:
  - `User#recent_posts`- lists 3 most recent posts
  - `Post#recent_comments`- lists 5 most recent comments
  - `Post#update_counter`- updates `posts_counter` of related user
  - `Like#update_counter`- updates `likes_counter` of related post
  - `Comment#update_counter`- updates `comments_counter` of related post

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  has_many :comments, inverse_of: 'post', dependent: :destroy
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_counter
+    count = author.posts_counter || 0
+    author.update_columns(posts_counter: count + 1)
+  end
```
Comment (gemmen29) [inline app/models/post.rb]: - [ ] [OPTIONAL] Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the update_counter method every time a new post has been saved to the database, then make the update_post_counter method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_counter
+    count = post.likes_counter || 0
+    post.update_columns(likes_counter: count + 1)
+  end
```
Comment (gemmen29) [inline app/models/like.rb]: - [ ] [OPTIONAL] Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the update_counter method every time a new post has been saved to the database, then make the update_post_counter method private

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_counter
+    count = post.comments_counter || 0
+    post.update_columns(comments_counter: count + 1)
+  end
```
Comment (gemmen29) [inline app/models/comment.rb]: - [ ] [OPTIONAL] Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the update_counter method every time a new post has been saved to the database, then make the update_post_counter method private

File: app/models/post.rb (hunk 1 above)
Comment (gemmen29) [inline app/models/post.rb]: - [ ] [OPTIONAL] Kindly update your counter code to be like this
```ruby
  def update_counter
    author.increment!(:posts_counter)
  end
```
as it's much more efficient, cleaner, and easier to read.

**Review by gemmen29 — APPROVED** [review]
Hi @IndieCoderMM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions
- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (IndieCoderMM) [reply app/models/post.rb]: Thanks for your insightful suggestions. 🙏😃 I'll make sure to implement these changes.

**Pull request merged**